### ISSUE TITLE
Stopped working on GNOME 45 (Fedora 39)

### ISSUE THREAD

**Issue by k2helix** [DimaZirix/fedora-gdm-wallpaper#7]
I just upgraded my system and running set-gdm-wallpaper returns an error that has to do with gnome-shell.css:

sed: can't read /tmp/tmp.nHrCcg4PDb/org/gnome/shell/theme/gnome-shell.css: No such file or directory

This was not happening when using Fedora 38.

**Comment by ghost** [comment]
On my setup I had to change the references of gnome-shell.css to gnome-shell-dark.css in the script.

**Comment by shayisnear** [comment]
same happening to me, sadly wont work.

**Comment by TylerCode** [comment]
> On my setup I had to change the references of gnome-shell.css to gnome-shell-dark.css in the script.

Has this also worked for Fedora 40?

**Comment by pihomeserver** [comment]
> > On my setup I had to change the references of gnome-shell.css to gnome-shell-dark.css in the script.
> 
> Has this also worked for Fedora 40?

Yes also on Fedora 42 / Gnome 48. Run `sudo vi /usr/sbin/set-gdm-wallpaper` then update line 116
`sed -i -z -E "s/#lockDialogGroup \{[^}]+/#lockDialogGroup \{$new_theme_params/g" "$workdir/org/gnome/shell/theme/gnome-shell-dark.css"`

**Comment by k2helix** [comment]
It's been a while since I opened this issue but I'm not sure if I should close it as it hasn't been really fixed without applying the changes from the previous comment. If anyone is looking for an alternative, take a look at [GDM Settings](https://github.com/gdm-settings/gdm-settings)
